tests/default.nix

```diff
@@ -0,0 +1,54 @@
+{
+  bash,
+  bats,
+  callPackage,
+  coreutils,
+  direnv,
+  fetchurl,
+  findutils,
+  gnugrep,
+  lib,
+  nix-direnv,
+  nixVersions,
+  writeShellScriptBin,
+}:
+let
+  direnv-stdlib = fetchurl {
+    url = "https://raw.githubusercontent.com/direnv/direnv/refs/tags/v2.37.1/stdlib.sh";
+    hash = "sha256-MMM04OXhqS/rRSuv8uh7CD70Z7CaGT63EtL/3LC08qM=";
+  };
+  bats-support = callPackage ./nix/bats-support.nix { };
+  bats-assert = callPackage ./nix/bats-assert.nix { };
+  mkTestRunner =
+    nixVersion:
+    writeShellScriptBin "test-runner-${nixVersion}" ''
+      set -e
+      export PATH=${
+        lib.makeBinPath [
+          bash
+          direnv
+          nixVersions.${nixVersion}
+          coreutils
+          findutils
+          gnugrep
+        ]
+      }
+      export DIRENV_STDLIB=${direnv-stdlib}
+      export DIRENVRC="${nix-direnv}/share/nix-direnv/direnvrc"
+      export BATS_LIB_PATH="${bats-support}:${bats-assert}"
+
+      echo run unittest
+      ${lib.getExe' bats "bats"} -x --verbose-run tests/
```

Comment:
Personally, I find the output here horribly unhelpful in the github interface. I left it like it was because I was able to test manually (and pretty reliably and quickly, since we're not actually gc'ing any more), but if others find it useless too, I'll remove `-x --verbose-run`.

Reply:
Honestly, I think that means we need to pull and run the commands locally. I shouldn't be debugging based on output from CI test runs, I don't think. And if I *must*, I think there's deeper problems that need to be addressed first. This is also partially why I removed the need to actually run a garbage collection. It was incredibly inconvenient for me, as I don't run gcs that frequently and don't make use of auto-gc, so the runs would take forever and undoubtedly get rid of something I wanted somewhere else.

You can also test with `bats ./tests` and should get the same output. I'm looking into why you might not.